simple_certmanager/models.py

```diff
@@ -16,6 +21,96 @@
+        max_length=100, help_text=_("The state or province name"), blank=True
+    )
+    email_address = models.EmailField(
+        help_text=_("Email address for the certificate"), blank=True
+    )
+    csr = models.TextField(
+        editable=False,
+        blank=True,
+        help_text=_("Certificate Signing Request"),
+        verbose_name=_("CSR"),
+    )
+    private_key = models.TextField(
+        editable=False,
+        blank=True,
+    )
+
+    def save(self, *args, **kwargs):
+        # Generate private key if not present
+        self.generate_private_key()
+
+        if not self.csr:
+            # Generate CSR if not present
+            csr_builder = x509.CertificateSigningRequestBuilder()
+            csr_builder = self.generate_csr(csr_builder)
+
+            # Load private key for signing CSR
+            private_key = serialization.load_pem_private_key(
+                self.private_key.encode(), password=None, backend=default_backend()
+            )
+
+            # Sign CSR with private key
+            csr = csr_builder.sign(private_key, hashes.SHA256(), default_backend())
+            # Only store the CSR bytes in the text field
+            self.csr = csr.public_bytes(serialization.Encoding.PEM).decode()
+
+        super().save(*args, **kwargs)
+
+    def generate_csr(self, csr_builder):
+        csr_builder = csr_builder.subject_name(
+            x509.Name(
+                [
+                    x509.NameAttribute(NameOID.COUNTRY_NAME, self.country_name),
+                    x509.NameAttribute(
+                        NameOID.STATE_OR_PROVINCE_NAME, self.state_or_province_name
+                    ),
+                    x509.NameAttribute(NameOID.LOCALITY_NAME, self.organization_name),
+                    x509.NameAttribute(
+                        NameOID.ORGANIZATION_NAME, self.organization_name
+                    ),
+                    x509.NameAttribute(NameOID.COMMON_NAME, self.common_name),
+                    x509.NameAttribute(NameOID.EMAIL_ADDRESS, self.email_address),
+                ]
+            )
+        )
+
+        return csr_builder
+
+    def generate_private_key(self):
+        if not self.private_key:
+            # Generate private key
+            private_key = rsa.generate_private_key(
+                public_exponent=65537, key_size=4096, backend=default_backend()
+            )
+            private_key_file_bytes = private_key.private_bytes(
+                encoding=serialization.Encoding.PEM,
+                format=serialization.PrivateFormat.PKCS8,
+                encryption_algorithm=serialization.NoEncryption(),
+            )
+            # Only store the private key bytes in the text field
+            self.private_key = private_key_file_bytes.decode()
+
+    def __str__(self):
+        return _("Signing Request #%(pk)s for %(common_name)s") % {
+            "pk": self.pk,
+            "common_name": self.common_name,
+        }
```

Comment:
can you use the `_("...").format(...)` string interpolation instead? The `%` formatting is legacy.